G/GMP/GMP@6.3.0/bundled/patches/gmp_alloc_overflow_func.patch

```diff
@@ -0,0 +1,111 @@
+diff --git a/errno.c b/errno.c
+index d71c146..40d83bd 100644
+--- a/errno.c
++++ b/errno.c
+@@ -38,8 +38,11 @@ see https://www.gnu.org/licenses/.  */
+ 
+ #include "gmp-impl.h"
+ 
++static void __gmp_default_alloc_overflow(void);
++
+ int gmp_errno = 0;
+ 
++static void   (*__gmp_alloc_overflow_func) (void) = __gmp_default_alloc_overflow;
+ 
+ /* Use SIGFPE on systems which have it. Otherwise, deliberate divide
+    by zero, which triggers an exception on most systems. On those
+@@ -72,6 +75,29 @@ __gmp_divide_by_zero (void)
+ }
+ void
+ __gmp_overflow_in_mpz (void)
++{
++  (*__gmp_alloc_overflow_func)();
++}
++
++static void
++__gmp_default_alloc_overflow(void)
+ {
+   __gmp_exception (GMP_ERROR_MPZ_OVERFLOW);
```

Comment:
So the overflow patch could be omitted if we instead caught SIGFPE and handled `gmp_errno` suitably.

Alternatively (and perhaps better), we could just use a single patch which adds an optional callback to `__gmp_exception`, and then it is up to us how want to handle the exception. Perhaps we could even convince the GMP folks to accept such a patch properly?